chain/store/index.go

```diff
@@ -27,8 +27,7 @@ func init() {
 }
 
 type ChainIndex struct {
-	indexCacheLk sync.Mutex
-	indexCache   map[types.TipSetKey]*lbEntry
+	indexCache *lru.ARCCache[types.TipSetKey, *lbEntry]
```

Comment:
We initially had an ARCCache here which was dropped in favor of a simple map with a mutex for the reasons pointed here: https://github.com/filecoin-project/lotus/pull/10423#pullrequestreview-1331509633
I think we can keep the mutex but reduce the size of critical section covered by the mutex

Reply:
It could be replaced by a simple LRU cache if that is a concern.